tests/topotests/ospf_single_switch/test_ospf_single_switch.py

```diff
@@ -0,0 +1,168 @@
+       ---              ---                               ---
+        |                |            OSPF-passive         |
+        |                |                                 |
+    +---+---+        +---+---+                         +---+---+
+    |       |        |       |                         |       |
+    |  r1   |        |  r2   |          (...)          |  rN   |
+    |       |        |       |                         |       |
+    +---+---+        +---+---+                         +---+---+
+        |                |           OSPF-active          |
+        |                |                                |
+        +----------------+--------------------------------+
+                             switch
+
+
+"""
+
+N_ROUTERS = 8
+
+# Save the Current Working Directory to find configuration files.
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, "../"))
+
+# Required to instantiate the topology builder class.
+pytestmark = [pytest.mark.ospfd, pytest.mark.bgpd]
+
+
+def build_topo(tgen):
+    "Build the topology"
+
+    # Create a single switch to connect all the routers
+    switch = tgen.add_switch("switch")
+
+    # Create routers
+    for router_id in range(1, N_ROUTERS + 1):
+        router = tgen.add_router(f"r{router_id}")
+        switch.add_link(router)
+
+        # The shared network needs to be connected to something
+        dummy = tgen.add_switch(f"s{router_id}")
+        dummy.add_link(router)
+
+
+def setup_module(mod):
+    logger.info("OSPF single switch:\n {}".format(TOPOLOGY))
+
+    tgen = Topogen(build_topo, mod.__name__)
+    tgen.start_topology()
+
+    # Starting Routers
+    router_list = tgen.routers()
+
+    for rname, router in router_list.items():
+        logger.info("Loading router %s" % rname)
+        router.load_frr_config(os.path.join(CWD, "{}_frr.conf".format(rname)))
+
+    # Initialize all routers.
+    tgen.start_router()
+
+
+def teardown_module(mod):
+    "Tear-down the test environment"
+    tgen = get_topogen()
+    tgen.stop_topology()
+
+
+def test_all_routes_advertised():
+    tgen = get_topogen()
+
+    if tgen.routers_have_failure():
+        pytest.skip("Skipped because of router(s) failure")
+
+    # networks advertised by each router and the expected next hops
+    networks = {
+        "r1": ("10.0.1.0/24", "120.0.0.1"),
+        "r2": ("10.0.2.0/24", "120.0.0.2"),
+        "r3": ("10.0.3.0/24", "120.0.0.3"),
+        "r4": ("10.0.4.0/24", "120.0.0.4"),
+        "r5": ("10.0.5.0/24", "120.0.0.5"),
+        "r6": ("10.0.6.0/24", "120.0.0.6"),
+        "r7": ("10.0.7.0/24", "120.0.0.7"),
+        "r8": ("10.0.8.0/24", "120.0.0.8"),
+    }
+
+    for router_orig in tgen.routers().keys():
+        for router_dest, network in networks.items():
+            if router_orig != router_dest:
+                check_route(router_orig, network[0], network[1])
+
+
+def check_route(router_name, network, expected_nexthop):
```

Comment:
Can't this be replaced to our existing function `verify_rib()`?

Reply:
I added the functions verify_rib() and verify_ospf_rib(), but they don't do what is required. They only check the presence of the expected route, but they don't check if they are active. As a result, they succeed in cases where check_route() fails. For now, I am leaving verify_rib(), verify_ospf_rib(), and check_route().